sdk/metrics/src/test/java/io/opentelemetry/sdk/metrics/AttributesAdviceTest.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.metrics;
+
+import static io.opentelemetry.api.common.AttributeKey.stringKey;
+import static io.opentelemetry.sdk.testing.assertj.OpenTelemetryAssertions.assertThat;
+import static io.opentelemetry.sdk.testing.assertj.OpenTelemetryAssertions.equalTo;
+import static java.util.Arrays.asList;
+
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.metrics.DoubleCounter;
+import io.opentelemetry.api.metrics.DoubleCounterBuilder;
+import io.opentelemetry.extension.incubator.metrics.ExtendedDoubleCounterBuilder;
+import io.opentelemetry.sdk.testing.exporter.InMemoryMetricReader;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+
+class AttributesAdviceTest {
+
+  private static final Attributes ATTRIBUTES =
+      Attributes.builder()
+          .put(stringKey("key1"), "1")
+          .put(stringKey("key2"), "2")
+          .put(stringKey("key3"), "3")
+          .build();
+
+  private SdkMeterProvider meterProvider = SdkMeterProvider.builder().build();
+
+  @AfterEach
+  void cleanup() {
+    meterProvider.close();
+  }
+
+  @Test
+  void counterWithoutAdvice() {
+    InMemoryMetricReader reader = InMemoryMetricReader.create();
+    meterProvider = SdkMeterProvider.builder().registerMetricReader(reader).build();
+
+    DoubleCounter counter =
+        meterProvider.get("meter").counterBuilder("counter").ofDoubles().build();
+    counter.add(1, ATTRIBUTES);
+
+    assertThat(reader.collectAllMetrics())
+        .satisfiesExactly(
+            metric ->
+                assertThat(metric)
+                    .hasDoubleSumSatisfying(
+                        sum -> sum.hasPointsSatisfying(point -> point.hasAttributes(ATTRIBUTES))));
+  }
+
+  @Test
+  void counterWithAdvice() {
```

Comment:
Hey so I made a [comment](https://github.com/open-telemetry/opentelemetry-specification/issues/3622#issuecomment-1654613883) over in the spec which states:

> Just so we're clear, my interpretation is that because advice is non-identifying, two histograms with the same identity and different advices will result in 1 metric stream. And thus we have to pick one set of advice to apply to the resulting metric stream, which should naturally be the first we see.

But that isn't the case as implemented today (and also not for the histogram bucket advice). You can demonstrate this by creating another counter with the same name, and with no advice, and seeing that there are two metrics produced instead of just one:
```
    DoubleCounter counter2 = meterProvider.get("meter").counterBuilder("counter").ofDoubles().build();
    counter2.add(1, ATTRIBUTES);
```

The issue stems from [MetricStorageRegistry#register(MetricStorage)](https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/state/MetricStorageRegistry.java#L57-L87), a tricky little piece of code responsible for evaluating whether a newly registered instrument should get a new storage (and thus a new metric stream at collection) or should send its measurements to an existing storage (and thus merge with an existing metric stream at collection).

The problem is that we use `MetricDescriptor.equals(..)` to determine if an existing storage matches, but that is a pretty strict equality, checking equality of name, description, view, InstrumentDescriptor (including name, description, unit, type, value type, and advice), and aggregation name. 

And I believe there's a bug where if a instrument matches except the case of the name, we produce two metric streams, instead of one like I thought we did. See [spec issue #3626](https://github.com/open-telemetry/opentelemetry-specification/pull/3626). 

I think this is probably fine for this PR. I need to go back and fix `MetricStorageRegistry#register` in a more general capacity.

Reply:
Opened #5683 to discuss.

Reply:
Yeah, I 100% agree advice should be non-identifying. 

> I think this is probably fine for this PR. I need to go back and fix `MetricStorageRegistry#register` in a more general capacity.

👍

Reply:
Solved in #5701.